src/algorithms/tonal/audio2midi.cpp

```diff
@@ -0,0 +1,143 @@
+#include "audio2midi.h"
+
+using namespace std;
+using namespace essentia;
+using namespace standard;
+
+const char *Audio2Midi::name = "Audio2Midi";
+const char *Audio2Midi::category = "Pitch";
+const char *Audio2Midi::description = DOC("Wrapper around Audio2Pitch and Pitch2Midi for real time application");
+
+void Audio2Midi::configure()
+{
+  _sampleRate = parameter("sampleRate").toReal();
+  // _frameSize = parameter("frameSize").toInt();
+  _hopSize = parameter("hopSize").toInt();
+  // _pitchAlgorithm = parameter("pitchAlgorithm").toString();
+  // _loudnessAlgorithm = parameter("loudnessAlgorithm").toString();
+  _minFrequency = parameter("minFrequency").toReal();
+  _maxFrequency = parameter("maxFrequency").toReal();
+  _tuningFrequency = parameter("tuningFrequency").toInt();
+  _pitchConfidenceThreshold = parameter("pitchConfidenceThreshold").toReal();
+  _loudnessThreshold = parameter("loudnessThreshold").toReal();
+  _transposition = parameter("transpositionAmount").toInt();
+  _minOccurrenceRate = parameter("minOccurrenceRate").toReal();
+  _midiBufferDuration = parameter("midiBufferDuration").toReal();
+  _minNoteChangePeriod = parameter("minNoteChangePeriod").toReal();
+  _minOnsetCheckPeriod = parameter("minOnsetCheckPeriod").toReal();
+  _minOffsetCheckPeriod = parameter("minOffsetCheckPeriod").toReal();
+  
+  // define frameSize depending on sampleRate
+  switch (static_cast<int>(_sampleRate)) {
+    case 16000:
+      _frameSize = 2048;
+      break;
+    case 24000:
+      _frameSize = 4096;
+      break;
+    case 44100:
+      _frameSize = _fixedFrameSize;
+      break;
+    case 48000:
+      _frameSize = _fixedFrameSize;
+      break;
+    default:
+      _frameSize = _fixedFrameSize;
+  }
+
+  _applyTimeCompensation = parameter("applyTimeCompensation").toBool();
+
+  if (_frameSize > _sampleRate * 0.5)
+  {
+    throw EssentiaException("Sax2Pitch: Frame size cannot be higher than Nyquist frequency");
```

Comment:
Clarify the text: different units of measurements compared. Unclear why this condition is required.

Reply:
I think is not needed if we define the frameSize internally, it is legacy code. Initially it was a sanity check when the frameSize was a pameter. I will remove this snippet.